crates/biome_css_analyze/src/lint/nursery/no_unknown_pseudo_class_selector.rs

```diff
@@ -0,0 +1,218 @@
+use crate::{
+    keywords::{WEBKIT_SCROLLBAR_PSEUDO_CLASSES, WEBKIT_SCROLLBAR_PSEUDO_ELEMENTS},
+    utils::{is_custom_selector, is_known_pseudo_class, is_page_pseudo_class, vendor_prefixed},
+};
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, RuleSource};
+use biome_console::markup;
+use biome_css_syntax::{
+    CssBogusPseudoClass, CssPageSelectorPseudo, CssPseudoClassFunctionCompoundSelector,
+    CssPseudoClassFunctionCompoundSelectorList, CssPseudoClassFunctionIdentifier,
+    CssPseudoClassFunctionNth, CssPseudoClassFunctionRelativeSelectorList,
+    CssPseudoClassFunctionSelector, CssPseudoClassFunctionSelectorList,
+    CssPseudoClassFunctionValueList, CssPseudoClassIdentifier, CssPseudoElementSelector,
+};
+use biome_rowan::{declare_node_union, AstNode, SyntaxNodeCast, TextRange};
+
+declare_rule! {
+    /// Disallow unknown pseudo-class selectors.
+    ///
+    /// For details on known pseudo-class, see the [MDN web docs](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-classes)
+    ///
+    /// This rule ignores vendor-prefixed pseudo-class selectors.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```css,expect_diagnostic
+    /// a:unknown {}
+    /// ```
+    ///
+    /// ```css,expect_diagnostic
+    /// a:UNKNOWN {}
+    /// ```
+    ///
+    /// ```css,expect_diagnostic
+    /// a:hoverr {}
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```css
+    /// a:hover {}
+    /// ```
+    ///
+    /// ```css
+    /// a:focus {}
+    /// ```
+    ///
+    /// ```css
+    /// :not(p) {}
+    /// ```
+    ///
+    /// ```css
+    /// input:-moz-placeholder {}
+    /// ```
+    ///
+    pub NoUnknownPseudoClassSelector {
+        version: "next",
+        name: "noUnknownPseudoClassSelector",
+        language: "css",
+        recommended: false,
```

Comment:
```suggestion
        recommended: true,
```

Reply:
I fixed it in https://github.com/biomejs/biome/pull/3034/commits/83e8e9f29c8234e5da8fdb75a4744984c49054b2